modules/environment/aws/iam/cert-manager/main.tf

```diff
@@ -26,34 +26,34 @@ resource "aws_iam_role" "cert_manager_service_account" {
 EOF
 }
 
-#tfsec:ignore:aws-iam-no-policy-wildcards
 resource "aws_iam_role_policy" "cert_manager" {
   name = "${var.cluster}-cert-manager"
   role = aws_iam_role.cert_manager_service_account.name
 
-  policy = <<EOF
-{
- "Version": "2012-10-17",
- "Statement": [
-    {
-        "Effect": "Allow",
-        "Action": "route53:GetChange",
-        "Resource": "arn:aws:route53:::change/*"
-    },
+  policy = jsonencode(
     {
-        "Effect": "Allow",
-        "Action":  [
-          "route53:ChangeResourceRecordSets",
-          "route53:ListResourceRecordSets"
-        ],
-        "Resource": "arn:aws:route53:::hostedzone/*"
-    },
-    {
-        "Effect": "Allow",
-        "Action": "route53:ListHostedZonesByName",
-        "Resource": "*"
-    }
- ]
-}
-EOF
+      "Version" : "2012-10-17",
+      "Statement" : [
+        # Cert-Manager can be limited in the access it has for each hosted zone that it touches,
+        #   but according to the docs[1], "cert-manager needs to be able to add records to 
```

Comment:
I think the wildcard is needed specifically to get a change to a record set that was already requested.  The permission below this one is for `ChangeResourceRecordSets`, which would be more in line with what your comment is describing.

In other words, this comment isn't wrong, but I don't think it explains why the wildcard is necessary.  There's no way for us to know the change ID when we apply this terraform, because those changes are happening at runtime.

Reply:
Ahh okay. I didn't see a very explicit explanation on the docs page that I found so this makes a lot of sense.